client/components/AdminChallengeEdit.js

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { connect } from 'react-redux';
+
+import { convertBufferToImgSrc } from '../utils';
+
+const _AdminChallengeEdit = props => {
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [difficulty, setDifficulty] = useState(1);
+  const [html, setHtml] = useState('');
+  const [css, setCss] = useState('');
+
+  // TODO upload image file, convert to binary, save in db
+  const [imageWidth, setImageWidth] = useState('');
+  const [imageHeight, setImageHeight] = useState('');
+  const [image, setImage] = useState({ data: '' });
+
+  useEffect(() => {
+    document.getElementById('name').focus();
```

Comment:
Could you explain to what is this line of code doing? .focus()

Reply:
Good question.  When the component loads, it puts the cursor in the `name` input field so that you do not have to do that manually.  Makes for a much better user experience.